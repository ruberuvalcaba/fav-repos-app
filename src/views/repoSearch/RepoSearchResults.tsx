import React from 'react'
import { Box, List, ListItem, Text } from '@chakra-ui/react'
import { Repo } from '../../types'
import { StarsCount } from '../sharedComponents'
interface SearchRepoResultsProps {
  results: Repo[]
  onSelect: (repo: Repo) => void
}

const SearchRepoResults = ({ results, onSelect }: SearchRepoResultsProps): JSX.Element => {
  return (
    <Box
      style={{
        height: 'calc(100vh - 165px)',
        overflow: 'auto',
        boxShadow: '0 2px 5px 0 rgba(0, 0, 0, 0.1)',
        borderBottomLeftRadius: '5px',
        borderBottomRightRadius: '5px',
      }}
    >
      <List spacing={3} py={3} px={3} bg="white">
        {results.map((repo: Repo) => {
          return (
            <ListItem
              key={repo.id}
              bg="gray.100"
              p={3}
              style={{
                cursor: 'pointer',
                borderRadius: '5px',
              }}
              _hover={{
                bg: 'gray.200',
              }}
              onClick={() => onSelect(repo)}
            >
              <Text fontSize="sm" color="blackAlpha.900">
                {repo.name}
              </Text>
              <Text fontSize="xs" color="blackAlpha.700">
                {repo.description}
              </Text>

              <Box>
                <Box display="flex" mt="2" alignItems="center" justifyContent="space-between">
                  <Text fontSize="xs" color="blue.500">
                    {repo.language}
                  </Text>
                  <StarsCount stars={repo.stargazersCount} />
                </Box>
              </Box>
            </ListItem>
          )
        })}
      </List>
    </Box>
  )
}

export default SearchRepoResults
