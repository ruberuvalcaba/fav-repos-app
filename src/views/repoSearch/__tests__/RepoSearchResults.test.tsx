import React from 'react'
import renderer from 'react-test-renderer'
import RepoSearchResults from '../RepoSearchResults'

it('RepoSearchResults component renders correctly', () => {
  const data = [
    {
      id: '73475252',
      name: 'react-native-calendars',
      fullName: 'wix/react-native-calendars',
      description: 'React Native Calendar Components',
      url: 'https://api.github.com/repos/wix/react-native-calendars',
      createdAt: new Date('2016-11-11T12:17:27Z'),
      stargazersCount: 8452,
      language: 'TypeScript',
    },
    {
      id: '36749153',
      name: 'react-native-code-push',
      fullName: 'microsoft/react-native-code-push',
      description: 'React Native module for CodePush',
      url: 'https://api.github.com/repos/microsoft/react-native-code-push',
      createdAt: new Date('2015-06-02T17:19:20Z'),
      stargazersCount: 8452,
      language: 'C',
    },
  ]
  const component = renderer.create(<RepoSearchResults results={data} onSelect={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
