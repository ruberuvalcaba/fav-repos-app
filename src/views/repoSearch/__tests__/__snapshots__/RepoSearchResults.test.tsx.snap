// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoSearchResults component renders correctly 1`] = `
<div
  className="css-0"
  style={
    {
      "borderBottomLeftRadius": "5px",
      "borderBottomRightRadius": "5px",
      "boxShadow": "0 2px 5px 0 rgba(0, 0, 0, 0.1)",
      "height": "calc(100vh - 165px)",
      "overflow": "auto",
    }
  }
>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-oqxdmq{list-style-type:none;padding-top:3px;padding-bottom:3px;-webkit-padding-start:3px;padding-inline-start:3px;-webkit-padding-end:3px;padding-inline-end:3px;background:white;}.css-oqxdmq>*:not(style)~*:not(style){margin-top:3px;}",
      }
    }
    data-emotion="css oqxdmq"
  />
  <ul
    className="css-oqxdmq"
    role="list"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-14bpjno{background:gray.100;padding:3px;}.css-14bpjno:hover,.css-14bpjno[data-hover]{background:gray.200;}",
        }
      }
      data-emotion="css 14bpjno"
    />
    <li
      className="css-14bpjno"
      onClick={[Function]}
      style={
        {
          "borderRadius": "5px",
          "cursor": "pointer",
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1hrv3v4{font-size:sm;color:blackAlpha.900;}",
          }
        }
        data-emotion="css 1hrv3v4"
      />
      <p
        className="chakra-text css-1hrv3v4"
      >
        react-native-calendars
      </p>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-ffgvoo{font-size:xs;color:blackAlpha.700;}",
          }
        }
        data-emotion="css ffgvoo"
      />
      <p
        className="chakra-text css-ffgvoo"
      >
        React Native Calendar Components
      </p>
      <div
        className="css-0"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1uru6j0{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-top:2px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;}",
            }
          }
          data-emotion="css 1uru6j0"
        />
        <div
          className="css-1uru6j0"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1m5fkwh{font-size:xs;color:blue.500;}",
              }
            }
            data-emotion="css 1m5fkwh"
          />
          <p
            className="chakra-text css-1m5fkwh"
          >
            TypeScript
          </p>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-70qvj9{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
              }
            }
            data-emotion="css 70qvj9"
          />
          <div
            className="css-70qvj9"
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-2n4y2f{display:inline-block;line-height:1em;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;color:gray.300;vertical-align:middle;width:4px;height:4px;margin-right:1px;}",
                }
              }
              data-emotion="css 2n4y2f"
            />
            <svg
              className="chakra-icon css-2n4y2f"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-fyr1uu{font-size:2xs;color:blackAlpha.700;}",
                }
              }
              data-emotion="css fyr1uu"
            />
            <p
              className="chakra-text css-fyr1uu"
            >
              8.5k
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      className="css-14bpjno"
      onClick={[Function]}
      style={
        {
          "borderRadius": "5px",
          "cursor": "pointer",
        }
      }
    >
      <p
        className="chakra-text css-1hrv3v4"
      >
        react-native-code-push
      </p>
      <p
        className="chakra-text css-ffgvoo"
      >
        React Native module for CodePush
      </p>
      <div
        className="css-0"
      >
        <div
          className="css-1uru6j0"
        >
          <p
            className="chakra-text css-1m5fkwh"
          >
            C
          </p>
          <div
            className="css-70qvj9"
          >
            <svg
              className="chakra-icon css-2n4y2f"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <p
              className="chakra-text css-fyr1uu"
            >
              8.5k
            </p>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
